src/java/org/apache/cassandra/cql3/QueryProcessor.java

```diff
@@ -428,11 +455,16 @@ public static Prepared parseAndPrepare(String query, ClientState clientState, bo
             qualifiedStatement.setKeyspace(clientState);
             keyspace = qualifiedStatement.keyspace();
         }
-
         // Note: if 2 threads prepare the same query, we'll live so don't bother synchronizing
         CQLStatement statement = raw.prepare(clientState);
         statement.validate(clientState);
 
+        // Set CQL string for AlterSchemaStatement as this is used to serialize the transformation
+        // in the cluster metadata log
+        if (statement instanceof AlterSchemaStatement)
+            ((AlterSchemaStatement)statement).setCql(query);
```

Comment:
would it be possible to serialize schema transformation in more formalized way? I don't know if it is possible to have bound variables in alter schema statement, but if so, or it will be possible in the future, what would happen then?